server/pulp/server/content/web/views.py

```diff
@@ -0,0 +1,143 @@
+import os
+
+from django.http import \
+    HttpResponse, HttpResponseRedirect, HttpResponseNotFound, HttpResponseForbidden
+from django.views.generic import View
+
+from pulp.common.config import parse_bool
+from pulp.repoauth.wsgi import allow_access
+from pulp.server.lazy import URL, Key
+from pulp.server.config import config as pulp_conf
+
+
+class ContentView(View):
+    """
+    The content delivery view provides content.
+
+    :ivar key: The public RSA key used for URL signing.
+    :type key: M2Crypto.RSA.RSA
+    """
+
+    @staticmethod
+    def urljoin(scheme, host, port, base, path, query):
+        """
+        Join a base URL and path component.
```

Comment:
This looks like a combination of ``urlunparse`` and ``urljoin`` from the ``urlparse`` module. Can you just call those two functions and let them handle the details?